internal/jimmhttp/http.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 Canonical.
+
+package jimmhttp
+
+import (
+	"context"
+	"net/http"
+)
+
+type HTTPHandler struct {
```

Comment:
This is called HTTPHandler but can only proxy according to its servehttp? This is very confusing?